### PR TITLE
Support `[P` escape (delete character)

### PR DESCRIPTION
**Pull request by dimapaloskin**
Hello.

I worked with bash output and faced with the next line:
`\r\u001b[C\u001b[C\u001b[2Pls`

There is `[2P` escape code that means "delete two characters".
As I can see for now it doesn't supported. 
Maybe there is a strong reason for skipping "P" char in the current regex. If so please explain my why :)

### CODE REVIEW

**Comment by sindresorhus** [comment]
// @Qix- @arjunmehta

File: fixtures/ansi-codes.js [hunk 1]
```diff
@@ -45,6 +45,9 @@ exports.ansiCompatible = {
 	'[J': ['Erase from cursor to end of screen'],
 	'[0J': ['Same'],
 	'[2J': ['Erase entire screen'],
+	'[P' : ['Delete charecter'],
```
Comment (Qix-) [inline fixtures/ansi-codes.js]: `Delete character`

File: fixtures/ansi-codes.js [hunk 2]
```diff
@@ -45,6 +45,9 @@ exports.ansiCompatible = {
...
+	'[0P': ['Same'],
```
Comment (Qix-) [inline fixtures/ansi-codes.js]: `Delete character (0P)`

File: index.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
 'use strict';
 module.exports = function () {
-	return /[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]/g;
+	return /[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-RZcf-nqry=><]/g;
```
Comment (Qix-) [inline index.js]: Can you instead change it to `A-PRZcf`? You're introducing `Q` as well, which hasn't been brought up as a valid escape sequence.

**Comment by Qix-** [comment]
Just curious, do you know what's causing the `P` output? I have never seen it before - it's undocumented.

**Comment by dimapaloskin** [comment]
Hello @Qix.

You can find description of the code here: [http://www.inwap.com/pdp10/ansicode.txt](http://www.inwap.com/pdp10/ansicode.txt). (312 Line, DCH). Also this document contains [Q description.

I faced with [P code while parsing bash output. In some cases while I browsing history (UP or DOWN keys) bash removes old output via this code

For example:
"ls" - old input buffer
"\u001b[2Pid" - new input buffer.

So on the monitor I can see "id" command instead "ls".

**Comment by Qix-** [comment]
Ah interesting. At first glance this looks like a standard proposal that was never really adopted. Thanks for the think; that's really useful.

Definitely okay with the nature of this PR. Let me take a second look.

**Comment by Qix-** [comment]
Published as `2.1.1`. Thanks again!

**Pull request merged**